Abracadabra/AbracadabraAPI/AbracadabraAPI/Controllers/UsersController.cs

```diff
@@ -49,9 +49,30 @@ public async Task<ActionResult<IEnumerable<UserViewModel>>> GetUsers([FromQuery]
             return viewModels;
         }
 
+        //GET: api/Users/{Username}
+        [HttpGet("byname/{Username}")]
+        [Authorize(Roles = "Admin")]
+        public async Task<ActionResult<IEnumerable<UserViewModel>>> GetUsersByName(string Username)
+        {
+            //current user so that the admin can't find himself and ban himself
+            var cuName = User.Identity.Name;
+
+            var users = await _userManager.Users.Where(u => u.NormalizedUserName.ToLower().Contains(Username.ToLower()) && u.UserName != cuName).ToListAsync();
+            List<UserViewModel> viewModels = new List<UserViewModel>();
+
+            foreach (ApplicationUser user in users)
+            {
+                var roles = await _userManager.GetRolesAsync(user);
+
+                 viewModels.Add(Mapper.UserToViewModel(user, roles[0]));

```

Comment:
Keep in mind that if the user is an expert in a specific subject it will not display his role as expert if you use _userManager to get the roles of the User.

Reply:
Okay we should look into this in the clean up sprint